src/java/com/unifina/feed/StreamrMessage.java

```diff
@@ -3,34 +3,17 @@
 import java.util.Date;
 import java.util.Map;
 
-import com.unifina.data.StreamrBinaryMessageV29.SignatureType;
 import com.unifina.feed.map.MapMessage;
 
 public class StreamrMessage extends MapMessage {
```

Comment:
I was kind of hoping that it might be possible to kill this class (and `MapMessage` and maybe even `ITimestamped`) and replace them with just `StreamMessage`. I'm not sure what kind of corner case problems you'll run into (maybe something with the time events). I recommend to leave this to be the last comment you address, and see if it is feasible.

Reply:
I tried but only managed to use only `StreamMessage` for field detection. Basically a lot of classes require `MessageClass` to implement `ITimestamped`. If `StreamMessage` was the only class used we could easily replace the generics with `MessageClass extends StreamMessage` but there is at least one other class used as `MessageClass`: `RuntimeRequest`.

Reply:
Yeah, I see! I can think of two quick solutions to make it work with the event queue (which assumes `ITimestamped`s) and still avoid duplicating things with `StreamrMessage`

- Wrap the `StreamMessage` in some class which implements `ITimestamped`:

```
public class WrappedStreamMessage implements ITimestamped {
    private final StreamMessage streamMessage
    public WrappedStreamMessage(StreamMessage streamMessage) {
        this.streamMessage = streamMessage;
    }
    // ...
}
```

- Or move the simple `ITimestamped` interface to the Java client and have `StreamMessage` implement it from the start!

I might go with the 2nd approach because it would allow us to use `StreamMessage`s in the event queue directly, and the `ITimestamped` interface obviously doesn't move any functionality to a wrong place.